capnp/lib/capnp.pyx

```diff
@@ -871,7 +871,7 @@ cdef _to_dict(msg, bint verbose, bint ordered):
                 ret[field] = _to_dict(temp_msg_b._get(field), verbose, ordered)
 
         return ret
-    elif msg_type is _DynamicStructReader:
+    elif msg_type is _DynamicStructReader or isinstance(msg, _Response):
         temp_msg_r = msg
         if ordered:
             ret = _collections.OrderedDict()
```

Comment:
To be honest I'm not sure what the check on line 892 is for at all. The infinite recursion would be from line 892 to `_DynamicStructReader.to_dict`, then immediately back to the `to_dict` function here. I'm not sure if the if-case on 892 is ever supposed to evaluate to True, or if these cases should always be handled by the respective if/elif on like 857/874. Otherwise the checks on 857/874 can just be the `isinstance` check.

Reply:
I'm less familiar with this part of the code, but it might have to do with Generics.